--- org/logmon.pl       2010-04-11 19:03:52.000000000 +0900
+++ logmon_20100411/logmon.pl   2014-01-03 22:08:40.644868514 +0900
@@ -73,19 +73,21 @@
 }

 sub watch_for {
-my ( $tail_num, $target, $message, $action, $new_action );
+my ( $tail_num, $target, $message, $action, $new_action , $line );

     ( $target, $tail_num ) = @_;
     unless ( fork() ) {      # Child
         $SIG{HUP}  = \&catch_hup;
         $SIG{TERM} = \&catch_term;
-        open ( IN, "tail -n$tail_num -f $target|" );
+        open ( IN, "tail -n$tail_num -F $target|" );
         while ( <IN> ) {
             foreach $message ( keys %{$Config{$target}} ) {
                 if ( $_ =~ m/$message/ ) {
                     foreach $action ( @{$Config{ $target }->{ $message }} ) {
+                        $line = $_;
+                        $line =~ s/\"/\\\"/g;
                         $new_action = $action;
-                        $new_action =~ s/<%%%%>/$_/g;
+                        $new_action =~ s/<%%%%>/$line/g;
                         system( $new_action );
                     }
                 }


